tests/ci/run_prefix_tests.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+set -exo pipefail
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+source tests/ci/common_posix_setup.sh
+
+print_system_and_dependency_information
+
+echo "Testing a prefix build of AWS-LC in debug mode."
+build_prefix_and_test
+
+echo "Testing a prefix build of AWS-LC in release mode."
+build_prefix_and_test -DCMAKE_BUILD_TYPE=Release
+
+echo "Testing a prefix build of AWS-LC small compilation."
+build_prefix_and_test -DOPENSSL_SMALL=1 -DCMAKE_BUILD_TYPE=Release
+
+echo "Testing a prefix build of AWS-LC in no asm mode."
+build_prefix_and_test -DOPENSSL_NO_ASM=1 -DCMAKE_BUILD_TYPE=Release
```

Comment:
Nit: Add one more dimension `-DASAN=1`? `ASAN`(`run_posix_sanitizer.sh`) maybe the most helpful dimension IMO. However, with that enabled, the ssl runner tests(8k) are very slow on arm when ASAN is enabled.

Reply:
I think we should consider a (prefix + ASAN) build test for a subsequent PR.